kong/dns/utils.lua

```diff
@@ -0,0 +1,287 @@
+local utils = require("kong.resty.dns.utils")
+
+local log = ngx.log
+
+local NOTICE = ngx.NOTICE
+
+local type = type
+local ipairs = ipairs
+local tonumber = tonumber
+local math_random = math.random
+local table_clear = require("table.clear")
+local table_insert = table.insert
+local table_remove = table.remove
+
+local readlines = require("pl.utils").readlines
+
+local DEFAULT_HOSTS_FILE = "/etc/hosts"
+local DEFAULT_RESOLV_CONF = "/etc/resolv.conf"
+
+local LOCALHOST = {
+  ipv4 = "127.0.0.1",
+  ipv6 = "[::1]",
+}
+
+local DEFAULT_HOSTS = { localhost = LOCALHOST }
+
+
+local _M = {}
+
+
+-- checks the hostname type
+-- @return "ipv4", "ipv6", or "name"
+function _M.hostname_type(name)
+  local remainder, colons = name:gsub(":", "")
+  if colons > 1 then
+    return "ipv6"
+  end
+
+  if remainder:match("^[%d%.]+$") then
+    return "ipv4"
+  end
+
+  return "name"
```

Comment:
```suggestion
  return "domain"
```

The term `name` seems a bit overloaded in the code, and it is not as clear as `domain`.